testing/test.ts

```diff
@@ -46,9 +46,12 @@ export class Test {
   constructor(public readonly options: ITestOptions, private readonly fn: () => any) {}
 
   public async run(ctx: IRunContext) {
-    let lastResult: IRunResult;
     const path = [...ctx.path, this.options.name];
+    if (!path.some(pathPart => pathPart.match(ctx.testNameMatcher))) {
```

Comment:
One other thing, it might be useful to filter on a test in a specific suite for when we programmatically create suites.

In that case, I'd be tempted to just join the path parts with " > " like we do in the logs then match that